contracts/Genesis.sol

```diff
@@ -173,26 +173,27 @@ contract Genesis is ERC721Pausable, VRFConsumerBase, Ownable {
     }
 
     /**
-     * Mint reserved gods
-     * @param to address to send the god to
-     * @param count number of gods to transfer
+     * Function to mint the reserved gods
+     * TODO Add a to address to send directly to a wallet
+     * TODO Maybe we should remove the count and have this function run once to make it more simple
+     * @param count number of gods to mint from the reserved pool
      */
-    // function mintReservedGods(address to, uint256 count) public onlyOwner {
-    //     require(
-    //         GenesisSupply.reservedGodsSupply() >=
-    //             count + GenesisSupply.reservedGodsTransfered.current(),
-    //         "Not enough reserved gods left"
-    //     );
-    //     // Here we don't need to increment counter and god supply counter because we already do in the constructor
-    //     // to not initialize the counters at 0
-    //     for (uint256 i = 0; i < count; i++) {
-    //         GenesisSupply.reservedGodsTransfered.increment();
-    //         GenesisSupply.mintGod(
-    //             to,
-    //             GenesisSupply.reservedGodsTransfered.current()
-    //         );
-    //     }
-    // }
+    function mintReservedGods(uint256 count) external onlyOwner {
+        require(
+            GenesisSupply(genesisSupplyAddress).reservedGodsCurrentIndex() +
```

Comment:
Each `GenesisSupply(genesisSupplyAddress)` increased the gas cost by some good factor, making a call cost minimally 2100 gas.

Couldn't just we forward everything to the supply contract?

At the very least, some refactoring should be done to go from 3 calls to 1 that returns max supply + actual index.

Reply:
Ok, I'll check this after the merge, good call, thx for pointing it out!